### PR TITLE
Add Vale rule to require title-case H1s

### PR DESCRIPTION
**Pull request by ddbeck**
The style guide calls for title case H1s, in contrast to sentence case H2s elsewhere. Speculatively, I've written a Vale rule to require this and fixed up the handful of cases where this convention wasn't already followed.

If this is desirable, I can create a corresponding rule for sentence case on H2+ headings.

Prompted by @karensawrey in https://github.com/buildkite/docs/pull/1615#discussion_r911041368.

### CODE REVIEW

**Comment by plaindocs** [comment]
I actually thought I'd done this already, but maybe it's one of the things I fiddled with and didn't push.

Yeah, let's do it for H2 as well. I'd also like to mark which rules in the style Guide are validated by linters, and try to slowly increase that number.

**Review by karensawrey — APPROVED** [review]

**Comment by ddbeck** [comment]
@plaindocs I'd like to do the other headings in incremental follow up PRs, if that's OK with you. A quick look shows that the H2s are a bit more complicated and will require some actual decision making, since they don't as widely follow the Chicago style.

**Comment by plaindocs** [comment]
Sure thing :heavy_check_mark:

File: pages/agent/v2/cli_artifact.md.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# buildkite-agent artifact
+# `buildkite-agent artifact`
```
[the remark this answers is not in the document]
Comment (ddbeck) [reply pages/agent/v2/cli_artifact.md.erb:1]: The alternatives here are:

- Write an exception for each of these (OK)
- Use comments to toggle off the linting for these headings (clunkier, but closely ties config to exceptions)
- Change the headings to something else (greatest effort required—not a fan)

Personally, I'm a fan of code in headings. Generally, my style preferences favor uniformity for maintainability (i.e., the less I have to think about it, the easier it is to maintain).

Finally, I'll note that there exist other headings with code already. As a temporary measure, we could adopt this change, then address the question of code-in-headings across all headings another time.

[the remark this answers is not in the document]
Comment (olyism) [reply pages/agent/v2/cli_artifact.md.erb:1]: hi @ddbeck @plaindocs personally I like inline code in headings. At times it's a necessity to describe the section.

I'm cool for this to go live. The aesthetic could be better but I don't want it to hold up the work or stifle maintainability.

Will take note to address this in the upcoming typography refresh. Thanks :)

**Pull request merged**